### PR TITLE
Added plugins.yml file with list of plugins

### LINKED ISSUE #16: Create JSON or YAML file with plugin names and SHA
This repo contains formulas for all _supported_ `kn` plugins by the Knative community.

Other tooling might need such a list, this issue / proposal is to create a YAML or JSON file that would contain the list of plugins names, repos, and latest version SHAs and have the formulas use this file, instead of embedding into code. 

The side effect would be that such a list would be generally useful for other tools, e.g., the VSCode plugin by @joshuawilson or documentation generation for the list of “approved plugins”.

Here is one proposal for this file:

```yaml
---
- source-kafka:
    - 0.21.0:
        - sha: 9a10e1b0ff9a035d88b6d69d5d7bf235de8a2b4b1002062688e104fe936f4161
        - repo: https://github.com/knative-sandbox/kn-plugin-source-kafka
- admin:
    - 0.21.0:
        - sha: 62009d94134abb23f208b725ba47eb5da5ca8bf55a554300576cbc595f111069
        - repo: https://github.com/knative-sandbox/kn-plugin-admin 
```

Including a `repo` field to allow for plugins that would be located outside of `knative-sandbox` since otherwise I would have to assume that’s where it’s coming from. 

We could call it `plugins.yaml` and add new versions entries for each plugin with a new release and new top level entries for additional plugins as we vet them.

WDYT Josh? Could you use this?

CC: @rhuss @navidshaikh

### PR DESCRIPTION
**Pull request by maximilien**
# Changes

1. added plugins_helper.rb that parses plugins.yml
2. added plugins_helper_test.rb
3. modified admin.rb and source-kafka.rb to use plugins.yml

This should be transparent to homebrew users

- :gift: Add new feature

/kind enhancement 

Fixes #16 

CC: @joshuawilson @rhuss

### CODE REVIEW

**Comment by joshuawilson** [comment]
So you are building the download link based on the values in the yaml, correct?

**Comment by maximilien** [comment]
> So you are building the download link based on the values in the yaml, correct?

Yes. I can simplify it so that the `plugin_helper.rb` computes the download URL. The YAML will have all info. 

The next step (different PR) is to automate the generation of this YAML with a simple list of the plugins to include. But that's for future.

Let me know if the current YAML is usable to you so I can get this reviewed and merged.

File: plugins.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
```
Comment (navidshaikh) [inline plugins.yml:1]: The cadence to update the repo would be per release, how about grouping per release and then it includes the plugins under it ? This way we'll have to add only one block (of release) per release than updating multiple.
```
---
v0.22.0:
   source-kafka:
     [...]
   admin:
     [...]
```

File: plugins.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+source-kafka:
+  homepage: https://github.com/knative-sandbox/kn-plugin-source-kafka
+  0.22.0:
+    shas:
+      darwin-amd64: 9a410c2062a351f778fcf84a85909d50a886dfc5daeb5554a16b03db201ee83c
+      linux-amd64: 650136bb614910c38aa3c0e12321bba2fda1df321746ad4a1f53f3f5d2229096
+      linux-arm64: 8c76ea635fc06919dd15657dc8565db15f97d2dbe7d30302eab5f9d54e73f26c
+      windows-amd64.exe: 0ae2686d47e95ccb0c3177af5f6d9f8368bc2d17a72d96ba033300a34881f836
+      linux-s390x: f30ff3127b2a9e91f9a943589eb778590a4528be93720eecf9cf67aebb778e33
+      linux-ppc64le: dbbf4794694c8e9d6967f72e189200b64a1ede09bebfd7e7a426570d91f9230e
```
Comment (navidshaikh) [inline plugins.yml:11]: these entries are not needed ?

Comment (maximilien) [reply plugins.yml:11]: OK fair. I'll remove

File: plugins.yml (hunk 1 above)
Comment (maximilien) [reply plugins.yml:1]: OK, but in this case we end up repeating the `homepage` in  each version...  I suppose a small price to pay. I could make it the download so it's pegged to the version but the homepage is also quite useful. I'll figure something

**Comment by maximilien** [comment]
@navidshaikh see update and please try it and let me know. Thx

**Comment by maximilien** [comment]
@navidshaikh is this ready to merge? I want to sync with 0.24.0 and also add the diag plugin. Thanks.

**Comment by rhuss** [comment]
@maximilien sorry, we missed the 0.24 train ;-( I can have a look at the PR, but you would need to resolve the conflicts first that occurred because of the latest release.

**Comment by maximilien** [comment]
No longer required

**Pull request closed without merging**